File: translation_output/no_test/magicoder/codenet/python/java/p02729.java
<｜begin▁of▁sentence｜>You are an exceptionally intelligent coding assistant that consistently delivers accurate and reliable responses to user instructions.

@@ Instruction
Translate the following python code to java and enclose your solution inside ```java```:
```
n,m = map(int,input().split())

even = n * (n-1) // 2
odd = m * (m-1) // 2

print(even + odd)
```

@@ Response
```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int m = Integer.parseInt(input[1]);

        long even = (long) n * (n-1) / 2;
        long odd = (long) m * (m-1) / 2;

        System.out.println(even + odd);
    }
}
```

In Java, we use the `Scanner` class to read input from the user. The `split(" ")` method is used to split the input string into an array of strings, where each string is a word in the input. The `Integer.parseInt(input[i])` method is used to convert each string in the array to an integer.

The `//` operator in Java is used for integer division, which is equivalent to the `//` operator in Python. However, Java requires explicit casting to `long` for integer division to work correctly with large numbers.

The `System.out.println()` method is used to print the result to the console.<｜end▁of▁sentence｜>

File: translation_output/no_test/llama2/codenet/java/python/p02729.py
<s> Translate the following java code to python and enclose your solution inside ```python```:
```
import java.util.Scanner;

public class p02729 {
	public static void main(String[] args){
		try(Scanner sc = new Scanner(System.in)) {
			

			int n = sc.nextInt();
			int m = sc.nextInt();

			int nComb = n * (n-1) / 2;
			int mComb = m * (m-1) / 2;
			
			System.out.println(nComb + mComb);
			
		}
	}

}
```

### 解答

```python
import sys

n = int(sys.stdin.readline())
m = int(sys.stdin.readline())

nComb = n * (n-1) / 2
mComb = m * (m-1) / 2

print(nComb + mComb)
```

### 解析

```
import java.util.Scanner;

public class p02729 {
	public static void main(String[] args){
		try(Scanner sc = new Scanner(System.in)) {
			

			int n = sc.nextInt();
			int m = sc.nextInt();

			int nComb = n * (n-1) / 2;
			int mComb = m * (m-1) / 2;
			
			System.out.println(nComb + mComb);
			
		}
	}

}
```

- 使用 `try-with-resources` 语句来声明 `Scanner` 对象，并在 `finally` 块中关闭 `Scanner` 对象。
- 使用 `Scanner` 对象读取输入，并将输入转换为整数。
- 使用 `nComb` 和 `mComb` 变量来计算 `nComb` 和 `mComb` 的值。
- 使用 `print` 语句输出 `nComb + mComb` 的值。

### 复杂度分析

- 时间复杂度：`O(n)`。
- 空间复杂度：`O(1)`。

### 相关题目

- [p02728](../p02728)
- [p02730](../p02730)
</s>
